test/schedulers.py

```diff
@@ -0,0 +1,50 @@
+import test_utils
+from torch.optim.lr_scheduler import _LRScheduler
+
+MIN_DELTA_TO_UPDATE_LR = 0.000001
+NUM_WARMUP_EPOCHS = 1
+
+
+class WarmupAndExponentialDecayScheduler(_LRScheduler):
+  """TODO(zcain): docstring."""
+  def __init__(self, optimizer, num_steps_per_epoch, divide_every_n_epochs=20, divisor=5):
+    self.num_steps_per_epoch = num_steps_per_epoch
+    self.divide_every_n_epochs = divide_every_n_epochs
+    self.divisor = divisor
+    self.previous_lr = -1
+    self.max_lr = optimizer.param_groups[0]['lr']
+    self.optimizer = optimizer
+
+  def get_lr(self, epoch, step):
+    # Warmup epoch. Ramp up learning rate from 0.0 to self.max_lr using
+    # a linear slope.
+    if epoch <= NUM_WARMUP_EPOCHS:
+      return min(self.max_lr,
+                 self.max_lr * ((step + 1.0) / self.num_steps_per_epoch))
+
+    # Normal epoch. Use an exponential decay determined by init params.
+    # NOTE: We assume 1-indexed epochs here.
+    return self.max_lr / (
+        self.divisor ** ((epoch - 1) // self.divide_every_n_epochs))
+
+  def step(self, epoch, step, summary_writer=None):
```

Comment:
Feels like the `summary_writer` should be a constructor parameter, more then a `step()` function one.

Reply:
I second this

Reply:
Agreed - made this change